### PR TITLE
add deactivation and filter support

### PR DESCRIPTION
**Pull request by Adam-W1**
Delete (deactivate) a tag
https://dluhcdigital.atlassian.net/browse/FS-3236
Implement tagging search and filtering
https://dluhcdigital.atlassian.net/browse/FS-3265

Fixes
https://dluhcdigital.atlassian.net/browse/FS-3259
https://dluhcdigital.atlassian.net/browse/FS-3257

### Change description
_A brief description of the pull request_

- [ ] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)
- [ ] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
_If manual testing is needed, give suggested testing steps_

### Screenshots of UI changes (if applicable)
image
image
image
image
image

### CODE REVIEW

File: app/assess/data.py [hunk 1]
```diff
@@ -115,6 +118,49 @@ def get_available_tags_for_fund_round(fund_id, round_id) -> List[Tag]:
...
 
 
+def get_tag_for_fund_round(fund_id, round_id, tag_id) -> List[Tag]:
+    endpoint = Config.ASSESSMENT_TAG_ENDPOINT.format(
+        fund_id=fund_id, round_id=round_id, tag_id=tag_id
+    )
+    response = get_data(endpoint)
+    if response is not None:
+        current_app.logger.info(f"tag returned: {len(response)}")
+        result = Tag.from_dict(response)
```
Comment (tferns) [inline app/assess/data.py:128]: This isn't returning `List[Tag]` - should this be nested in a list?

File: app/assess/templates/create_tag.html [hunk 2]
```diff
@@ -54,7 +53,7 @@ <h1 class="govuk-fieldset__heading">
                         <input class="govuk-radios__input" id="{{form.type.id}}-{{tag_type['purpose']}}" name="{{form.type.id}}" type="radio" value="{{tag_type['id']}}" aria-describedby="tag-item-hint">
                         <label class="govuk-label govuk-radios__label" for="{{form.type.id}}-{{tag_type['purpose']}}">
                             {{tag_type['purpose'].title()}}
-                                <strong class="govuk-tag govuk-!-margin-left-2 govuk-tag--{{ tag_config[tag_type['purpose']]['colour'] }} govuk-tag">{{ tag_config[tag_type['purpose']]['colour'] if tag_config[tag_type['purpose']]['colour'].lower() != "grey" else "white" }}</strong>
+                                <strong class="govuk-tag govuk-!-margin-left-2 govuk-tag--{{ tag_config[tag_type['purpose']]['colour'] }} dluhc-tag">{{ tag_config[tag_type['purpose']]['colour'].title() if tag_config[tag_type['purpose']]['colour'].lower() != "grey" else "White" }}</strong>
```
Comment (tferns) [inline app/assess/templates/create_tag.html:56]: Does capitalisation matter here? `white`  - maybe not

File: app/assess/templates/deactivate_tag.html [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      <div class="govuk-form-group">
+        <fieldset class="govuk-fieldset" aria-describedby="deactivate-hint">
+          <div class="govuk-checkboxes" data-module="govuk-checkboxes">
+            <div class="govuk-checkboxes__item">
+              <input class="govuk-checkboxes__input" id="deactivate" name="deactivate" type="checkbox"
+                value="{{tag.id}}">
+              <label class="govuk-label govuk-checkboxes__label" for="deactivate">
+                I understand this tag will be removed from all {{tag.tag_association_count}} applications and this
+                <b>cannot</b> be undone
+              </label>
```
Comment (tferns) [inline app/assess/templates/deactivate_tag.html:34]: Do we need to show if there's no associations? or `not tag.tag_association_count`

Comment (tferns) [reply app/assess/templates/deactivate_tag.html:34]: Also single applications will be plural, might be nice to handle?

```jinja2
{% if tag.tag_association_count == 1 %}
    I understand this tag will be removed from 1 application, and this
{% elif tag.tag_association_count > 1 %}
    I understand this tag will be removed from all {{tag.tag_association_count}} applications, and this
{% endif %}
```

**Review by tferns — DISMISSED** [review]
Woah! Hefty amount of work, nice job.  Just a few minor comments, but looks good 👍

File: app/assess/data.py (hunk 1 above)
Comment (tferns) [reply app/assess/data.py:128]: Or the type annotation and `return []` need updated?

File: app/assess/templates/deactivate_tag.html (hunk 3 above)
Comment (Adam-W1) [reply app/assess/templates/deactivate_tag.html:34]: good shout 👍

File: app/assess/data.py (hunk 1 above)
Comment (Adam-W1) [reply app/assess/data.py:128]: good spot

**Pull request merged**